modules/core/src/v2/baseCoin.ts

```diff
@@ -115,6 +115,15 @@ export interface SupplementGenerateWalletOptions {
   };
   rootPrivateKey?: string;
   disableKRSEmail?: boolean;
+  addressDerivationKeypair?: {
+    encryptedPrv: string;
+    pub: string;
+  };
+}
+
+export interface DeriveKeypairOptions {
+  addressDerivationPrv: string;
+  index: number;
 }
```

Comment:
lets make `prv` an optional string and add required string `pub`.

for SOL, we can only do hardened derivation right now, but can change in the future. (Sol's impl can throw error if prv is not present)
other coins may not need to rely on hardened derivation.